src/common/monitors.cpp

```diff
@@ -1,84 +1,99 @@
 #include "pch.h"
 
-#include "common.h"
 #include "monitors.h"
 
+#include <Shellscalingapi.h>
+
+#include "common.h"
+
 bool operator==(const ScreenSize& lhs, const ScreenSize& rhs)
 {
     auto lhs_tuple = std::make_tuple(lhs.rect.left, lhs.rect.right, lhs.rect.top, lhs.rect.bottom);
     auto rhs_tuple = std::make_tuple(rhs.rect.left, rhs.rect.right, rhs.rect.top, rhs.rect.bottom);
     return lhs_tuple == rhs_tuple;
 }
 
-static BOOL CALLBACK get_displays_enum_cb(HMONITOR monitor, HDC hdc, LPRECT rect, LPARAM data)
+static BOOL CALLBACK getDisplaysEnumCb(HMONITOR monitor, HDC hdc, LPRECT rect, LPARAM data)
 {
-    MONITORINFOEX monitor_info;
-    monitor_info.cbSize = sizeof(MONITORINFOEX);
-    GetMonitorInfo(monitor, &monitor_info);
-    reinterpret_cast<std::vector<MonitorInfo>*>(data)->emplace_back(monitor, monitor_info.rcWork);
+    MONITORINFOEX monitorInfo;
+    monitorInfo.cbSize = sizeof(MONITORINFOEX);
+    if (GetMonitorInfo(monitor, &monitorInfo))
+    {
+        reinterpret_cast<std::vector<MonitorInfo>*>(data)->emplace_back(monitor, monitorInfo.rcWork);
+    }
     return true;
```

Comment:
We should return false if `GetMonitorInfo` failed.

Reply:
returning false here means enumeration will stop.
Returning true will continue enumeration without adding current monitor that failed.
Are you sure retrunign false is correct behaviour?

Reply:
If we fail to get the DPI scaling of a monitor we should assume it may have different DPI scaling, correct?

Reply:
Since it's a winapi callback from which we're returning to, checking `GetMonitorInfo` return code and skipping `emplace_back` if it failed seems like a correct thing to do. `MonitorInfo` struct contains only screen size info and monitor handle. Perhaps we could make screen size member optional, if we still want the handle to be present.

Reply:
I changed the method to handle dpi without need to GetMonitorInfo